### PR TITLE
[6580] Implement funding upload flow (without saving)

### PR DESCRIPTION
**Pull request by darokel**
### Context

This PR consolidates 4 trello tickets which cover similar work/logic for different types of funding uploads. 

- https://trello.com/c/1ifBcxNh/6580-funding-upload-form-lead-school-upload-trainee-summary
- https://trello.com/c/Cmr13rvI/6605-funding-upload-form-lead-school-upload-payment-schedule
- https://trello.com/c/7TAHXjcI/6606-funding-upload-form-providers-upload-payment-schedule
- https://trello.com/c/o9tDB6SQ/6607-funding-upload-form-providers-upload-trainee-summary

### Changes proposed in this pull request

- Renames boilerplate path from `upload_funding` to `funding_uploads` so it aligns with the resource/model
- Implement form submission page for each of the funding upload type
- Basic validations 
- Confirmation page

### Guidance to review

Dev: Review from commit https://github.com/DFE-Digital/register-trainee-teachers/pull/3930/commits/4530a4fb4b0d868e872ca99902a6fad249a4c4da onwards

Product: Head to the funding upload page and go through each of the funding upload type. Check if the flow makes sense?

### Notes

- In the prototype, there was a middle semi confirmation page for the file uploaded but after speaking with @andrewscrivener we have removed this to simplify things
- This PR only implements the form submission but the actual processing/saving logic is dealt with in an upcoming ticket

### CODE REVIEW

**Comment by darokel** [comment]
[Funding upload form: Lead school upload trainee summary](https://trello.com/c/1ifBcxNh/6580-funding-upload-form-lead-school-upload-trainee-summary)

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -17,8 +17,8 @@ def can_view_funding?
     FeatureService.enabled?("funding") && defined?(current_user) && !current_user&.system_admin
   end
 
-  def can_view_upload_funding?
-    FeatureService.enabled?("upload_funding") && defined?(current_user) && current_user&.system_admin
+  def can_view_funding_uploads?
```
Comment (carlosmartinez) [inline app/helpers/users_helper.rb:20]: is this still used? if this is within the system admin area, the authorisation will be covered by the route constraint.

File: app/views/system_admin/funding_uploads/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      <%= f.hidden_field :funding_type, value: funding_type %>
+
+      <h1 class="govuk-heading-l"><%= t("components.page_titles.funding_uploads.new.#{funding_type}") %></h1>
+
+      <p class="govuk-body"><%= t(".#{funding_type}") %></p>
+
+      <ol class="govuk-list govuk-list--number">
+        <li>
+          <%= f.govuk_select :month,
+              options_for_select([["Choose month", nil]] + Date::MONTHNAMES.compact.map.with_index(1) { |month, index| [month, index] }, selected: f.object.month),
```
Comment (carlosmartinez) [inline app/views/system_admin/funding_uploads/new.html.erb:21]: could chuck `[["Choose month", nil]] + Date::MONTHNAMES.compact.map.with_index(1) { |month, index| [month, index] }` in a helper

File: app/views/system_admin/funding_uploads/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+              label: { text: "Select applicable month" },
+              hint: { text: "Please select the month you are uploading funding information for." } %>
+        </li>
+        <li>
+          <%= f.govuk_file_field(:file, label: { text: "Upload the CSV file" }, accept: ".csv") %>
```
Comment (carlosmartinez) [inline app/views/system_admin/funding_uploads/new.html.erb:26]: does `accept` 'accept' file extensions or do we need a mime type here?

**Comment by carlosmartinez** [comment]
looks great, couple of minor points/questions

File: app/helpers/users_helper.rb (hunk 1 above)
Comment (darokel) [reply app/helpers/users_helper.rb:20]: Ah wait, this is to feature flag the funding upload stuff specifically from support users as it's incomplete

Comment (carlosmartinez) [reply app/helpers/users_helper.rb:20]: ah ok. maybe we can just inline the `FeatureService.enabled?("upload_funding")` then?

File: app/views/system_admin/funding_uploads/new.html.erb (hunk 3 above)
Comment (darokel) [reply app/views/system_admin/funding_uploads/new.html.erb:26]: Yeah it'll only whitelist the listed filetypes

File: app/views/system_admin/funding_uploads/new.html.erb (hunk 2 above)
Comment (darokel) [reply app/views/system_admin/funding_uploads/new.html.erb]: done

File: app/helpers/users_helper.rb (hunk 1 above)
Comment (darokel) [reply app/helpers/users_helper.rb]: Wait sorry it's definitely redundant. We do the same thing but for any feature flag with an existing method https://github.com/DFE-Digital/register-trainee-teachers/blob/99693c05125ff0d20561c047a45709e7e4386b2e/app/controllers/system_admin/funding_uploads_controller.rb#L5 which calls https://github.com/DFE-Digital/register-trainee-teachers/blob/3424e4731bb61e5f89de82f1c5de698b30fcc3d7/app/controllers/application_controller.rb#L126. Not sure why it was added, have removed it now.

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**